Must change changes to the checklist criteria (#132)
ansible-collections/overview: collection_requirements.rst

Why is this needed
===================

Collection Infrastructure
=========================

* MUST have a publicly available issue tracker, that does not require a paid level of service to create an account or view issues.

> * MUST have a CoC (or if not fall back to Ansible's: https://docs.ansible.com/ansible/latest/community/code_of_conduct.html )

Reviewer: ```suggestion
* MUST have a CoC (or if not fall back to Ansible's: https://docs.ansible.com/ansible/latest/community/code_of_conduct.html )
```
i'd avoid abbreviations

Reviewer 2: The discussion on CoC in the end went more in the direction "all collections in Ansible are governed by Ansible's CoC".

CI Testing
===========

> At a minimum ``ansible-test sanity`` MUST be run from the `latest stable ansible-base/ansible-core branch <https://github.com/ansible/ansible/branches/all?query=stable->`_. We suggest to *additionally* run ``ansible-test sanity`` from the ansible/ansible ``devel`` branch so that you find out about new linting requirements earlier, as well as all version of ``ansible-base``/``ansible-core`` that the collection supports.

Reviewer 3: ```suggestion
At a minimum ``ansible-test sanity`` MUST be run from the `latest stable ansible-base/ansible-core branch <https://github.com/ansible/ansible/branches/all?query=stable->`_. We suggest to *additionally* run ``ansible-test sanity`` from the ansible/ansible ``devel`` branch so that you find out about new linting requirements earlier, as well as all versions of ``ansible-base``/``ansible-core`` that the collection supports.
```

> At a minimum ``ansible-test sanity`` MUST be run from the `latest stable ansible-base branch <https://github.com/ansible/ansible/branches/all?query=stable->`_. We suggest to *additionally* run ``ansible-test sanity`` from the ansible/ansible ``devel`` branch so that you find out about new linting requirements earlier, as well as all version of ``ansible-base`` that the collection supports. For example ``plugins/modules/docker_container.py use-argspec-type-path # uses colon-separated paths, can't use type=path``

Reviewer 2: ```suggestion
At a minimum ``ansible-test sanity`` MUST be run from the `latest stable ansible-base/ansible-core branch <https://github.com/ansible/ansible/branches/all?query=stable->`_. We suggest to *additionally* run ``ansible-test sanity`` from the ansible/ansible ``devel`` branch so that you find out about new linting requirements earlier, as well as all version of ``ansible-base``/``ansible-core`` that the collection supports.
```

For most repository GitHub actions are sufficient, see `example <https://github.com/ansible-collections/collection_template/tree/main/.github/workflows>`_

> The sanity tests MUST pass but adding some tests to the ignore.txt file is an allowed method of getting them to pass.  All entries in ignores.txt MUST have a justification in a comment in the ignore.txt file for each entry.  Reviewers can block acceptance of a new collection if they don't agree with the ignores.txt entries.

Reviewer 2: ```suggestion
The sanity tests MUST pass but adding some entries to the ignore.txt file is an allowed method of getting them to pass.  All entries in ignores.txt MUST have a justification in a comment in the ignore.txt file for each entry.  For example ``plugins/modules/docker_container.py use-argspec-type-path # uses colon-separated paths, can't use type=path``. Reviewers can block acceptance of a new collection if they don't agree with the ignores.txt entries.
```

> All CI test MUST run against every PR & commit to the repo.

Reviewer 3: ```suggestion
All CI tests MUST run against every PR & commit to the repo.
```

> All CI tests MUST run against at nightly to ensure that repos without regular commits are tested against latest version of ansible-test from each ansible-base version tested. `Ansible Collection Template <https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml>`_. includes an example of how to achieve this

Reviewer 2: ```suggestion
All CI tests MUST run regularly (nightly, or at least once per week) to ensure that repos without regular commits are tested against latest version of ansible-test from each ansible-base/ansible-core version tested. `Ansible Collection Template <https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml>`_. includes an example of how to achieve this.
```

> All CI tests MUST run regularly (nightly, or at least once per week) to ensure that repos without regular commits are tested against latest version of ansible-test from each ansible-base/ansible-core version tested. `Ansible Collection Template <https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml>`_. includes an example of how to achieve this.

Reviewer 3: ```suggestion
All CI tests MUST run regularly (nightly, or at least once per week) to ensure that repos without regular commits are tested against the latest version of ansible-test from each ansible-base/ansible-core version tested. `Ansible Collection Template <https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml>`_. includes an example of how to achieve this.
```

CI Testing
===========

* You MUST run ``ansible-test sanity`` from the `latest stable ansible-base/ansible-core branch <https://github.com/ansible/ansible/branches/all?query=stable->`_. 
* You MUST run CI against all versions of ``ansible-base``/``ansible-core`` that the collection supports.
* You SHOULD suggest to *additionally* run ``ansible-test sanity`` from the ansible/ansible ``devel`` branch so that you find out about new linting requirements earlier.
* The sanity tests MUST pass.

  * Adding some entries to the ``test/sanity/ignore*.txt`` file is an allowed method of getting them to pass. 
  * All entries in ignores.txt MUST have a justification in a comment in the ignore.txt file for each entry.  For example ``plugins/modules/docker_container.py use-argspec-type-path # uses colon-separated paths, can't use type=path``.

>   * Reviewers can block acceptance of a new collection if they don't agree with the ignores.txt entries.

Reviewer 4: Hi guys, I couldn't make it at the meeting today. I think _some_ tests for which there should be no excuses: param type not in docs, or list missing elements type. Those are way too simple, and they are the single two major offenders in the ignore files today.

Author: yes, I think we agreed in theory to that but decided we could start approve new collections before there was a list of tests which must not be ignored.  If you'd like to propose which tests cannot be ignored, that would be great (use a separate PR and then put it on the meeting agenda).

* All CI tests MUST run against every PR & commit to the repo.
* All CI tests MUST run regularly (nightly, or at least once per week) to ensure that repos without regular commits are tested against the latest version of ansible-test from each ansible-base/ansible-core version tested. 

> All of the above can be achieved by using the following GitHub Action template , see `example <https://github.com/ansible-collections/collection_template/tree/main/.github/workflows>`_

Reviewer 2: ```suggestion
All of the above can be achieved by using the following GitHub Action template, see `example <https://github.com/ansible-collections/collection_template/tree/main/.github/workflows>`_.
```

Reviewer 5: Good spot, thanks @felixfontein
